layer/ro_layer.go

```diff
@@ -4,16 +4,24 @@ import (
 	"fmt"
 	"io"
 
+	"github.com/docker/distribution"
 	"github.com/docker/distribution/digest"
 )
 
+var _ ForeignSourcer = &roLayer{}
+
+func (rl *roLayer) ForeignSource() *distribution.Descriptor {
+	return rl.foreignSrc
+}
+
 type roLayer struct {
 	chainID    ChainID
 	diffID     DiffID
 	parent     *roLayer
 	cacheID    string
 	size       int64
 	layerStore *layerStore
+	foreignSrc *distribution.Descriptor
```

Comment:
No. We have a cache of digests for push/pull, but we do not store actual descriptor objects.